File: frontend/src/lib/types/work.ts
export interface Work {
  id: string;
  title: string;
  description: string;
  thumbnail: string;
  category: string;
  featured?: boolean;
  technologies: string[];
  github_link?: string | null;
  demo_link?: string | null;
  blog_link?: string | null;
  screenshots?: Record<string, unknown> | null;
  duration?: string | null;
  role?: string | null;
  learnings?: string | null;
}

File: backend/app/domain/entity/work.py
from pydantic import BaseModel
from typing import Optional


class Work(BaseModel):
    id: str
    title: str
    description: str
    thumbnail: str
    category: str
    featured: Optional[bool] = False
    technologies: list[str]
    github_link: Optional[str] = None
    demo_link: Optional[str] = None
    blog_link: Optional[str] = None
    screenshots: Optional[dict] = None
    duration: Optional[str] = None
    role: Optional[str] = None
    learnings: Optional[str] = None
